b2b-sample/guardio-insurance-sample-application/components/settingsComponents/identity-providers/idpDetailsSections/settings.js

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2022 WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *http://www.apache.org/licenses/LICENSE-2.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useCallback, useEffect, useState } from 'react';
+import { Form } from 'react-final-form';
+import { Button, ButtonToolbar, Loader, useToaster } from 'rsuite';
+import FormSuite from 'rsuite/Form';
+import { LOADING_DISPLAY_BLOCK, LOADING_DISPLAY_NONE } from '../../../../util/util/frontendUtil/frontendUtil';
+import { errorTypeDialog, successTypeDialog } from '../../../util/dialog';
+
+import styles from '../../../../styles/Settings.module.css';
+import decodeGetFederatedAuthenticators from
+    '../../../../util/apiDecode/settings/identityProvider/decodeGetFederatedAuthenticators';
+import decodeUpdateFederatedAuthenticators from
+    '../../../../util/apiDecode/settings/identityProvider/decodeUpdateFederatedAuthenticators';
+import { checkIfJSONisEmpty } from '../../../../util/util/common/common';
+import SettingsFormSelection from './settingsFormSection/settingsFormSelection';
+
+export default function General(props) {
+
+    const [loadingDisplay, setLoadingDisplay] = useState(LOADING_DISPLAY_NONE);
+    const [federatedAuthenticators, setFederatedAuthenticators] = useState({});
+
+    const toaster = useToaster();
+
+    const fetchData = useCallback(async () => {
+
+        const res = await decodeGetFederatedAuthenticators(
+            props.session, props.idpDetails.id, props.idpDetails.federatedAuthenticators.defaultAuthenticatorId
+        );
+        await setFederatedAuthenticators(res);
+    }, [props])
+
+    useEffect(() => {
+        fetchData();
+    }, [fetchData]);
+
+    const validate = values => {
+        let errors = {}
+        if (federatedAuthenticators.properties) {
+            federatedAuthenticators.properties.filter((property) => {
+                if (!eval(property.key).value) {
+                    errors[property.key] = 'This field cannot be empty';
+                }
+            })
+        }
+
+        return errors
+    }
+
+    const onDataSubmit = (response, form) => {
+        if (response) {
+            successTypeDialog(toaster, "Changes Saved Successfully", "Idp updated successfully.");
```

Comment:
`Changes are saved successfully`
`Idp is updated successfully`

Reply:
Fixed with https://github.com/wso2/samples-is/pull/416/commits/130d957728499e5e0e731eba0b663086f931fe95